[Product editor block]: Rename the Add button to Publish

octaedro opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR renames the `Add` button to `Publish` in the new editor.

Before

![Screenshot 2024-02-08 at 10 41 16](https://github.com/woocommerce/woocommerce/assets/1314156/aacd0ba0-1df2-48bf-92ea-2f850de34665)

After

![Screenshot 2024-02-08 at 12 17 50](https://github.com/woocommerce/woocommerce/assets/1314156/3f26523f-8ba7-4253-903f-eff30bf65822)


Closes #44456.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure the New product editor is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to Products > Add New
3. Add a product `Name` and press `Publish`.
4. Verify the `Publish` button works as expected.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

octaedro committed:
Rename Publish button

octaedro committed:
Fix e2e tests

octaedro committed:
Add changelog

octaedro committed:
Add WC changelog

louwie17 reviewed (changes requested):
Nice work, you just missed one place within the notice, which still says `Product added` we did want to say `Product published` -> https://github.com/woocommerce/woocommerce/blob/a8e4b27df2acc4bcee1e677142dcbd6a80424bca/packages/js/product-editor/src/components/header/publish-button/publish-button.tsx#L45

octaedro committed:
Modify "Product published" copy

octaedro commented:
Thank you @louwie17 for your review. I addressed the suggested change and modified the e2e test to handle that. Could you take another look at this PR?

louwie17 reviewed (approved):
LGTM 🚀
